### PR TITLE
File upload work in progress

### PR DESCRIPTION
**Pull request by nategbrown9**
This is work in progress on results file upload. For now this is an android only implementation, utilizing WorkManager for running in the background. It uses the new cross platform FileSystem api that is being developed as part of Okio, so parts of it should be able to eventually be used on iOS.

### CODE REVIEW

File: bridge-client/src/androidMain/kotlin/org/sagebionetworks/bridge/kmm/shared/upload/UploadManager.kt [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            resource.loadResource<UploadFile>()?.let {
+                processUploadFile(it)
+            }
+        }
+        //Check that all uploads succeeded and have been removed
+        return database.getResources(ResourceType.FILE_UPLOAD).isEmpty()
+    }
+
+    private suspend fun processUploadFile(uploadFile: UploadFile) {
+        //Does upload order matter to Bridge?
```
Comment (syoung-smallwisdom) [inline bridge-client/src/androidMain/kotlin/org/sagebionetworks/bridge/kmm/shared/upload/UploadManager.kt]: This looks like a TODO? Comment, sign, and date.

File: bridge-client/src/androidMain/kotlin/org/sagebionetworks/bridge/kmm/shared/upload/UploadManager.kt [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        )
+        //TODO: Need to think through error handling if we are unable to get an UploadSession -nbrown 12/16/20
+        return updatedResource.loadResource<UploadSession>()
+    }
+
+    private suspend fun loadRemoteUploadSession(uploadRequest: UploadRequest): String {
+        return Json.encodeToString(uploadsApi.requestUploadSession(uploadRequest))
+    }
+
+    @OptIn(ExperimentalFileSystem::class)
```
Comment (syoung-smallwisdom) [inline bridge-client/src/androidMain/kotlin/org/sagebionetworks/bridge/kmm/shared/upload/UploadManager.kt:99]: Have you tested this on iOS at all? just curious as to how experimental it is. ;)

File: bridge-client/src/androidMain/kotlin/org/sagebionetworks/bridge/kmm/shared/upload/UploadRequester.kt [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        )
+
+        return uploadFile
+    }
+
+    @OptIn(ExperimentalFileSystem::class)
+    private fun getFile(filename: String): Path {
+        val pathString = context.filesDir.absolutePath + Path.DIRECTORY_SEPARATOR + filename
+        return pathString.toPath(Path.DIRECTORY_SEPARATOR)
+
```
Comment (syoung-smallwisdom) [inline bridge-client/src/androidMain/kotlin/org/sagebionetworks/bridge/kmm/shared/upload/UploadRequester.kt]: Nit: empty line after return and before curly braces.

**Review by syoung-smallwisdom — APPROVED** [review]
Looks good. Couple of questions/nits.

File: bridge-client/src/androidMain/kotlin/org/sagebionetworks/bridge/kmm/shared/upload/UploadManager.kt (hunk 2 above)
Comment (nategbrown9) [reply bridge-client/src/androidMain/kotlin/org/sagebionetworks/bridge/kmm/shared/upload/UploadManager.kt:99]: I have not tested this on iOS yet. The folks at Square added this to Okio in December, and it currently is only available in snapshot developer builds.

**Pull request merged**